guava/src/com/google/common/escape/CharEscaperBuilder.java

```diff
@@ -108,9 +113,13 @@ public CharEscaperBuilder addEscapes(char[] cs, String r) {
    *
    * @return a "sparse" array that holds the replacement mappings.
    */
+  @SuppressWarnings(value = {"array.length.negative",//int max is constant -1, max + 1 = 0 is non negative
```

Comment:
Why can't you annotate `max` as `@GTENegativeOne` or (if it's really a constant) `@IntVal(1)`?